article/Rougier-2017.md

```diff
@@ -0,0 +1,132 @@
+---
+Title: "Weighted Voronoi Stippling"
+Author:
+  - name: Nicolas P. Rougier
+    affiliation: 1,2,3
+    ORCID: 0000-0003-0330-9428
+Address:
+  - code:    1
+    address: INRIA Bordeaux Sud-Ouest, Bordeaux, France.
+  - code:    2
+    address: LaBRI, Université de Bordeaux, Institut Polytechnique de Bordeaux,
+             Centre National de la Recherche Scientifique, UMR 5800, Talence, France.
+  - code:    3
+    address: Institut des Maladies Neurodégénératives, Université  de Bordeaux,
+             Centre National de la Recherche Scientifique, UMR 5293, Bordeaux, France.
+    
+Contact:
+  - Nicolas.Rougier@inria.fr
+Editor:
+  - Name Surname
+Reviewer:
+  - Name Surname
+  - Name Surname
+Publication:
+  received:  Feb, 1, 2017
+  accepted:  Feb, 1, 2017
+  published: Feb, 1, 2017
+  volume:    "**3**"
+  issue:     "**1**"
+  date:      Feb 2017
+Repository:
+  article:   "http://github.com/rescience/rescience-submission/article"
+  code:      "http://github.com/rescience/rescience-submission/code"
+  data:      "http://github.com/rescience/rescience-submission/data"
+  notebook:  
+Reproduction:
+  - "Weighted Voronoi Stippling, Adrian Secord. In: Proceedings of the 2Nd
+     International Symposium on Non-photorealistic Animation and
+     Rendering. NPAR ’02. ACM, 2002, pp. 37– 43."
+Bibliography:
+  Rougier-2017.bib
+---
+
+# Introduction
+
+Adrian Secord introduces in [@Secord:2002] a *techniques for generating stipple
+drawings from grayscale images using weighted centroidal Voronoi diagrams* as
+in *the traditional artistic technique of stippling that places small dots of
+ink onto paper such that their density give the impression of tone*. The paper
+is not accompanied by any code and, even though
+the [webpage](http://www.mrl.nyu.edu/~ajsecord/stipples.html) of the author
+points to a code archive, this one is actually nowhere to be found. We decided
+to replicate the method because it can be used for different purpose, especially
+in computational biology or neuroscience where the distribution of a cell
+population can be conveniently described by shades of grey. It is to be noted
+that since the publication of this paper, a number of related techniques have
+been
+proposed [@AscencioLopez:2010][@Balzer:2009a][@Balzer:2009b][@Bridson:2007]
+that may be more efficient but with lower quality. We thus privileged this
+implementation that is simple and provides high stippling quality.
+
+# Methods
+
+We applied the method proposed in the original paper with some variations due
+to the non-use of the GPU for computing the Voronoi diagram. In the original
+article, author computes the Voronoi diagram by taking advantage of the graphic
+card (GPU) and the fact that a set of cones seen from above actually represents
+a Voronoi diagram (see [@Hoff:1999]). We use instead
+the [QHull](http://www.qhull.org) library (through
+the [Scipy](https://scipy.github.io) Python package) for computing the Voronoi
```

Comment:
GPU approach would require opening a GL context and I thought it would be too much trouble to make the script work on OSX/Linux/Windows. Plus, the GPU suffers from lack of precision because it works at image level (even if authors use higher resolution to cope with such imprecision). Since the qhull library is directly accessible from within scipy, I prefer to use this.

I will add a justification in the text.

As for optimization, I implemented the optimized version that is less readable in my opinion. I'll leave the "old" code for informational purpoose.